web_responsive_pwa/views/webclient_templates.xml

```diff
@@ -0,0 +1,68 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <template id="web_layout_pwa" name="Web layout PWA" inherit_id="web.layout">
+        <xpath expr="//meta[@name='viewport']" position="after">
+            <!-- Add link rel manifest -->
+            <link rel="manifest" href="/web_responsive_pwa/static/src/json/manifest.json"/>
+            <!-- Add iOS meta tags and icons -->
+            <meta name="apple-mobile-web-app-capable" content="yes"/>
+            <meta name="apple-mobile-web-app-status-bar-style" content="black"/>
+            <meta name="apple-mobile-web-app-title" content="Odoo PWA"/>
+            <link rel="apple-touch-icon" href="/web_responsive_pwa/static/img/icons/icon-152x152.png"/>
+            <!-- Add meta theme-color -->
+            <meta name="theme-color" content="#2E69B5" />
+        </xpath>
+    </template>
+    <template id="assets_backend" name="PWA assets" inherit_id="web.assets_backend">
+        <xpath expr="." position="inside">
+            <script type="text/javascript" src="/web_responsive_pwa/static/src/js/pwa_install.js"/>
+        </xpath>
+    </template>
+    <template id="service_worker" name="PWA service worker">
+'use strict';
+const CACHE_NAME = '<t t-esc="pwa_cache_name"/>';
+const FILES_TO_CACHE = [
+  <t t-foreach="pwa_files_to_cache" t-as="file_to_cache">
+      '<t t-esc="file_to_cache"/>',
+  </t>
+];
```

Comment:
Can't you just use a normal JS file for JS code and download this data from a normal JSON controller? :thinking:

Reply:
I think @simahawk was suggesting something similar, but I am not sure this would work. More info about service workers: https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API/Using_Service_Workers

Anyway if someone provides a patch , I test it gladly.

Reply:
@Yajo I discussed that already w/ @eLBati  (elsewhere) and he was not convinced it was going to work.
IMO we could even avoid service worker if it's just for caching because is not required despite you found docs only related to it. See https://developer.mozilla.org/en-US/docs/Web/API/Cache
> You don't have to use it in conjunction with service workers, even though it is defined in the service worker spec.

In any case, we could add a huge TODO in the roadmap and approve what we get here as the alpha stage of this thing.

Reply:
The TODO option seems OK for now :blush:

Reply:
Done